relay/http.go

```diff
@@ -113,13 +114,15 @@ func (h *HTTP) Stop() error {
 func (h *HTTP) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	start := time.Now()
 
-	if r.URL.Path == "/ping" && (r.Method == "GET" || r.Method == "HEAD") {
-			w.Header().Add("X-InfluxDB-Version", "relay")
-			w.WriteHeader(http.StatusNoContent)
-			return
+	reqPath := r.URL.Path
+
+	if reqPath == "/ping" && (r.Method == "GET" || r.Method == "HEAD") {
+		w.Header().Add("X-InfluxDB-Version", "relay")
+		w.WriteHeader(http.StatusNoContent)
+		return
 	}
 
-	if r.URL.Path != "/write" {
+	if reqPath != "/write" && reqPath != "/query" {
```

Comment:
This should now be handled with a request router, and has gone grown far beyond a simple path check.